splash/har/qt.py

```diff
@@ -129,16 +133,17 @@ def reply2har(reply, include_content=False, binary_content=False):
         res["redirectURL"] = ""
 
     if include_content:
-        data = bytes(reply.readAll())
-        if binary_content:
-            res["content"]["encoding"] = "binary"
-            res["content"]["text"] = data
-            res["content"]["size"] = len(data)
-        else:
-            res["content"]["encoding"] = "base64"
-            res["content"]["text"] = base64.b64encode(data)
-            res["content"]["size"] = len(data)
-
+        content = getattr(reply, 'content', None)
+        if content is not None:
+            res["content"]["size"] = content.size()
+            bytes_array = bytes(content)
+
+            if not is_binary_string(bytes_array):
+                res["content"]["text"] = bytes_array.decode('utf8', 'replace')
```

Comment:
According to specs using base64 everywhere is fine. There are 2 issues with this auto-detection: 1) data loss and 2) more code is required at client. Base64 encoding solves both, even if it is not as nice. I think Chrome may be decoding data based on Content-Type headers, but I'm not sure. We should really get this right or leave base64 as-is; data loss is not acceptable. 

What's your use case? If it is machine which processes HAR files then using base64 everywhere simplifies code - you don't have to handle a case with non-base64 encoding.